zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -78,6 +83,62 @@ use input_selection::{
     GreedyInputSelector, GreedyInputSelectorError, InputSelector, InputSelectorError,
 };
 
+/// Describes the key inputs and UFVK for an account that was derived from a ZIP-32 HD seed and account index.
+#[derive(Debug, Clone)]
+pub struct HDSeedAccount(
+    pub HdSeedFingerprint,
+    pub zip32::AccountId,
+    pub UnifiedFullViewingKey,
+);
+
+/// Represents an arbitrary account for which the seed and ZIP-32 account ID are not known
+/// and may not have been involved in creating this account.
+#[derive(Debug, Clone)]
+pub enum ImportedAccount {
+    /// An account that was imported via its full viewing key.
+    Full(UnifiedFullViewingKey),
+}
+
+/// Describes an account in terms of its UVK or ZIP-32 origins.
+#[derive(Debug, Clone)]
+pub enum Account {
```

Comment:
nit: is there specific utility in having distinct `HdSeedAccount` and `ImportedAccount` types, or could the `Account` enum just use structured variants as I'd done in https://github.com/zcash/librustzcash/pull/1202/commits/688c36166a51d3120049fd011aff10481ba34cd3#diff-dc0076ef8eaf064e73a80f1a6210a9b41da40eb49b24cefdf00b2797a255abafL56-L94?

Reply:
Yes. By declaring distinct types, a function that is guaranteed to return an `HdSeedAccount` can declare that it does so in its signature. In fact IIRC there is at least one function (possibly inside the test code) that needed to do this to avoid its caller having to use a match expression to get at what surely must be inside.

Also, `ImportedAccount` isn't simply a UFVK. It's an enum that has a UFVK for now, and later will have other variants added.